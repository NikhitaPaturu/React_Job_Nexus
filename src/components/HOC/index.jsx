import React from 'react';
import PropTypes from 'prop-types';
import { LOGIN } from '../../routes/constants';

const higherOrderComponent = (WrappedComponent) => {
  class HOC extends React.Component {
      static propTypes = {
        login: PropTypes.shape({})
      }

      static defaultProps = {
          login: false
      }
    
    componentDidMount() {
        const { login: { authorized }, history } = this.props;
        if(!authorized) {
            history.replace(LOGIN);
            return;
        }
    }
    render() {
      return <WrappedComponent {...this.props} />;
    }
  }
  return HOC;
};

export default higherOrderComponent;