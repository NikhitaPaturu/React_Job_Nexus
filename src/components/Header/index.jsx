import React, { Component } from 'react'
import { Menu } from 'semantic-ui-react'
import PropTypes from 'prop-types';
import './style.css';

export default class Header extends Component {
  state = { activeItem: 'home' }

  constructor(props) {
    super(props);
    this.handleItemClick = this.handleItemClick.bind(this);
  }
  
  static propTypes = {
      authorized: PropTypes.bool,
  }

  static defaultProps = {
    authorized: false,
  }

  handleItemClick = (e, { name }) => {
      this.setState({ activeItem: name })
      if (name === 'logout') {
          this.props.logoutHandler(false);
          this.props.history.push('/login');
      } else if (name === 'register') {
          this.props.history.push('/register');
      }
      else if (name === 'login') {
        this.props.history.push('/login');
    }
  }


  render() {
    const { activeItem } = this.state
    const { authorized } = this.props

    return (
        <div className="app-header"> 
            <p className="app-title">Nexus</p>
            <span>odd jobs platform</span>  
            <Menu inverted>
                <Menu.Item name='home' active={activeItem === 'home'} onClick={this.handleItemClick} />
                <Menu.Item
                name='contact'
                active={activeItem === 'contact'}
                onClick={this.handleItemClick}
                />
                { !authorized && <Menu.Menu position="right">
                        <Menu.Item
                        icon="user"
                        name='register'
                        active={activeItem === 'register'}
                        onClick={this.handleItemClick}
                        />
                        <Menu.Item
                        icon="sign-in alternate"
                        name='login'
                        active={activeItem === 'login'}
                        onClick={this.handleItemClick}
                        />
                    </Menu.Menu>
                }

                { authorized && <Menu.Menu position="right">
                        <Menu.Item
                        name='my account'
                        active={activeItem === 'my account'}
                        onClick={this.handleItemClick}
                        />
                        <Menu.Item
                        icon="sign-out alternate"
                        name='logout'
                        active={activeItem === 'logout'}
                        onClick={this.handleItemClick}
                        />
                    </Menu.Menu>
                }
            </Menu>
        </div>
    )
  }
}