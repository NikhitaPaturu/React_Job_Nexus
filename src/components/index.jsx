export { default as Header } from './Header';
export { default as Toggle } from './Toggle';
export { default as Page } from './Page';
export { default as HOC } from './HOC';