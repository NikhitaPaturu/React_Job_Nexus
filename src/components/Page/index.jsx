import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';
import './style.css';
import { Header } from '../../components';

export default class Page extends React.PureComponent {

    static propTypes = {
        authorized: PropTypes.bool,
        children: PropTypes.node,
        logoutHandler: PropTypes.func,
        history: PropTypes.shape({}).isRequired,
    }

    static defaultProps = {
        authorized: false,
        children: {},
        logoutHandler: noop
    }

    render() {
        const { authorized, children, logoutHandler, history } = this.props;

        return (
            <div className='app-container'>
                <Header authorized={authorized} logoutHandler={logoutHandler} history={history}></Header>
                {children}
            </div>
        )
    }
};