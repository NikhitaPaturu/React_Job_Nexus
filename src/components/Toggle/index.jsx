import React from 'react';
import PropTypes from 'prop-types';
import { Header } from '../../components';

export default class Toggle extends React.PureComponent {
    state = {
        showLabel : true
    }

    static propTypes = {
        title: PropTypes.string
    }

    static defaultProps = {
        title: ''
    }

    toggleLabel = () => {
        this.setState({ showLabel: !this.state.showLabel });
    }

    render() {
        const { title, ...obj } = this.props;

        return (
        <div>
            { title && <Header title={title} {...obj}/>}
            <button onClick={this.toggleLabel} {...obj} >Click Me! </button>
            {this.state.showLabel && <p>Hello World</p> }
        </div>
        )
    }
};