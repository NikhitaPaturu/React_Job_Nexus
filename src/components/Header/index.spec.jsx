import React from 'react';
import { shallow, mount } from 'enzyme';
import { Header } from '../../components';

const history = {
  push: jest.fn()
}

const logoutHandler = jest.fn()

describe('@render', () => {
    it('should render as expected if not authorized', () => {
      const component = shallow(<Header authorized={false} history={history}></Header>);

      expect(component).toMatchSnapshot();
    });

    it('should render as expected if authorized', () => {
      const component = shallow(<Header authorized={true} history={history}></Header>);

      expect(component).toMatchSnapshot();
    });
})

describe('@click', () => {
  it('simulate register click on register', () => {
    const component = mount(<Header authorized={false} history={history}></Header>);
    component.find('a').at(2).simulate('click');
    
  });

  it('simulate register click on login', () => {
    const component = mount(<Header authorized={false} history={history}></Header>);
    component.find('a').at(3).simulate('click');
  });

  it('simulate register click on logout', () => {
    const component = mount(<Header authorized={true} history={history} logoutHandler={logoutHandler}></Header>);
    component.find('a').at(3).simulate('click');
    
  });
})