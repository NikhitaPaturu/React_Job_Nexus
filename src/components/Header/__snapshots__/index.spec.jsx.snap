// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@render should render as expected if authorized 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    authorized={true}
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="app-title"
>
          Nexus
</p>,
        <span>
          odd jobs platform
</span>,
        <Menu
          inverted={true}
>
          <MenuItem
                    active={true}
                    name="home"
                    onClick={[Function]}
          />
          <MenuItem
                    active={false}
                    name="contact"
                    onClick={[Function]}
          />
          <MenuMenu
                    position="right"
          >
                    <MenuItem
                              active={false}
                              name="my account"
                              onClick={[Function]}
                    />
                    <MenuItem
                              active={false}
                              icon="sign-out alternate"
                              name="logout"
                              onClick={[Function]}
                    />
          </MenuMenu>
</Menu>,
      ],
      "className": "app-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Nexus",
          "className": "app-title",
        },
        "ref": null,
        "rendered": "Nexus",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "odd jobs platform",
        },
        "ref": null,
        "rendered": "odd jobs platform",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuItem
              active={true}
              name="home"
              onClick={[Function]}
/>,
            <MenuItem
              active={false}
              name="contact"
              onClick={[Function]}
/>,
            false,
            <MenuMenu
              position="right"
>
              <MenuItem
                            active={false}
                            name="my account"
                            onClick={[Function]}
              />
              <MenuItem
                            active={false}
                            icon="sign-out alternate"
                            name="logout"
                            onClick={[Function]}
              />
</MenuMenu>,
          ],
          "inverted": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "name": "home",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "name": "contact",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <MenuItem
                  active={false}
                  name="my account"
                  onClick={[Function]}
/>,
                <MenuItem
                  active={false}
                  icon="sign-out alternate"
                  name="logout"
                  onClick={[Function]}
/>,
              ],
              "position": "right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "name": "my account",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "icon": "sign-out alternate",
                  "name": "logout",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="app-title"
>
            Nexus
</p>,
          <span>
            odd jobs platform
</span>,
          <Menu
            inverted={true}
>
            <MenuItem
                        active={true}
                        name="home"
                        onClick={[Function]}
            />
            <MenuItem
                        active={false}
                        name="contact"
                        onClick={[Function]}
            />
            <MenuMenu
                        position="right"
            >
                        <MenuItem
                                    active={false}
                                    name="my account"
                                    onClick={[Function]}
                        />
                        <MenuItem
                                    active={false}
                                    icon="sign-out alternate"
                                    name="logout"
                                    onClick={[Function]}
                        />
            </MenuMenu>
</Menu>,
        ],
        "className": "app-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Nexus",
            "className": "app-title",
          },
          "ref": null,
          "rendered": "Nexus",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "odd jobs platform",
          },
          "ref": null,
          "rendered": "odd jobs platform",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuItem
                active={true}
                name="home"
                onClick={[Function]}
/>,
              <MenuItem
                active={false}
                name="contact"
                onClick={[Function]}
/>,
              false,
              <MenuMenu
                position="right"
>
                <MenuItem
                                active={false}
                                name="my account"
                                onClick={[Function]}
                />
                <MenuItem
                                active={false}
                                icon="sign-out alternate"
                                name="logout"
                                onClick={[Function]}
                />
</MenuMenu>,
            ],
            "inverted": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": true,
                "name": "home",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "name": "contact",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <MenuItem
                    active={false}
                    name="my account"
                    onClick={[Function]}
/>,
                  <MenuItem
                    active={false}
                    icon="sign-out alternate"
                    name="logout"
                    onClick={[Function]}
/>,
                ],
                "position": "right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "name": "my account",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "icon": "sign-out alternate",
                    "name": "logout",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`@render should render as expected if not authorized 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    authorized={false}
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="app-title"
>
          Nexus
</p>,
        <span>
          odd jobs platform
</span>,
        <Menu
          inverted={true}
>
          <MenuItem
                    active={true}
                    name="home"
                    onClick={[Function]}
          />
          <MenuItem
                    active={false}
                    name="contact"
                    onClick={[Function]}
          />
          <MenuMenu
                    position="right"
          >
                    <MenuItem
                              active={false}
                              icon="user"
                              name="register"
                              onClick={[Function]}
                    />
                    <MenuItem
                              active={false}
                              icon="sign-in alternate"
                              name="login"
                              onClick={[Function]}
                    />
          </MenuMenu>
</Menu>,
      ],
      "className": "app-header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Nexus",
          "className": "app-title",
        },
        "ref": null,
        "rendered": "Nexus",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "odd jobs platform",
        },
        "ref": null,
        "rendered": "odd jobs platform",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <MenuItem
              active={true}
              name="home"
              onClick={[Function]}
/>,
            <MenuItem
              active={false}
              name="contact"
              onClick={[Function]}
/>,
            <MenuMenu
              position="right"
>
              <MenuItem
                            active={false}
                            icon="user"
                            name="register"
                            onClick={[Function]}
              />
              <MenuItem
                            active={false}
                            icon="sign-in alternate"
                            name="login"
                            onClick={[Function]}
              />
</MenuMenu>,
            false,
          ],
          "inverted": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": true,
              "name": "home",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "name": "contact",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <MenuItem
                  active={false}
                  icon="user"
                  name="register"
                  onClick={[Function]}
/>,
                <MenuItem
                  active={false}
                  icon="sign-in alternate"
                  name="login"
                  onClick={[Function]}
/>,
              ],
              "position": "right",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "icon": "user",
                  "name": "register",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "active": false,
                  "icon": "sign-in alternate",
                  "name": "login",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          false,
        ],
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="app-title"
>
            Nexus
</p>,
          <span>
            odd jobs platform
</span>,
          <Menu
            inverted={true}
>
            <MenuItem
                        active={true}
                        name="home"
                        onClick={[Function]}
            />
            <MenuItem
                        active={false}
                        name="contact"
                        onClick={[Function]}
            />
            <MenuMenu
                        position="right"
            >
                        <MenuItem
                                    active={false}
                                    icon="user"
                                    name="register"
                                    onClick={[Function]}
                        />
                        <MenuItem
                                    active={false}
                                    icon="sign-in alternate"
                                    name="login"
                                    onClick={[Function]}
                        />
            </MenuMenu>
</Menu>,
        ],
        "className": "app-header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Nexus",
            "className": "app-title",
          },
          "ref": null,
          "rendered": "Nexus",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "odd jobs platform",
          },
          "ref": null,
          "rendered": "odd jobs platform",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <MenuItem
                active={true}
                name="home"
                onClick={[Function]}
/>,
              <MenuItem
                active={false}
                name="contact"
                onClick={[Function]}
/>,
              <MenuMenu
                position="right"
>
                <MenuItem
                                active={false}
                                icon="user"
                                name="register"
                                onClick={[Function]}
                />
                <MenuItem
                                active={false}
                                icon="sign-in alternate"
                                name="login"
                                onClick={[Function]}
                />
</MenuMenu>,
              false,
            ],
            "inverted": true,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": true,
                "name": "home",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "name": "contact",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <MenuItem
                    active={false}
                    icon="user"
                    name="register"
                    onClick={[Function]}
/>,
                  <MenuItem
                    active={false}
                    icon="sign-in alternate"
                    name="login"
                    onClick={[Function]}
/>,
                ],
                "position": "right",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "icon": "user",
                    "name": "register",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "icon": "sign-in alternate",
                    "name": "login",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            false,
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
